ServiceApp/src/org/droidplanner/core/mission/Mission.java

```diff
@@ -265,9 +269,13 @@ public void onMissionLoaded(List<msg_mission_item> msgs) {
     }
 
     private List<MissionItem> processMavLinkMessages(List<msg_mission_item> msgs) {
+        waypointToMissionItem.clear();
```

Comment:
@ne0fhyk I could dump the data from into into temporary variables and them compare to see if I should overwrite what's already on tower. Unless the missions each have some UID (which I doubt) which can be used to identify them.